c-runtime/Makefile

```diff
@@ -118,7 +118,12 @@ test: curl_catch
 .PHONY: test_clean
 test_clean:
 	make -f $(TESTDIR)/Systemtests/makefile clean
-	$(RM) -r $(TESTDIR)/include/catch.hpp
+
+	$(RM) -r $(TESTDIR)/include/catch_amalgamated.hpp
+	$(RM) -r $(TESTDIR)/src/catch_amalgamated.cpp
+
+	touch $(TESTDIR)/include/catch_amalgamated.hpp
+	touch $(TESTDIR)/src/catch_amalgamated.cpp
```

Comment:
Wozu dienen diese touches? Da zu vorher die Dateien gelöscht hast, erzeugt das doch einfach zwei leeree Dateien?

Ich fände es sinnvoller, die beiden catch-dateien wirklich zu löschen und auf .gitignore zu setzen. Wenn sie nicht existieren, muss das Makefile vor dem Kompilieren den Download anstoßen, d.h. `test` muss einfach nur von `curl_catch` abhängen (was es bereits tut). Das funktioniert aber nur, wenn da nicht zwei leere Dateien mit nem frischen Zeitstempel liegen!

Reply:
Das Problem ist, dass ich dann in der makefile Variable C_TEST_FILES die beiden Namen nicht drin habe. Deswegen müssen die Dateien schon vorher existieren. Ich könnte vielleicht beim Downloaden die Namen hinzufügen, aber da sie so oder so da sein müssen ist das touch ein anderer Weg dafür. Ergibt das Sinn?

Reply:
Nochmal anders, wenn die touches weg sind wird diese Variable ohne die catch_....cpp Datei erstellt und undefined errors werden geworfen. So gehe ich sicher, dass die Datei vorher schon existiert, auch wenn sie nur leer ist.

Reply:
Eigentlich reicht auch ein touch auf nur die .cpp Datei

Reply:
Ein Problem mit den Timestamps habe ich jetzt nicht beim wiederholten Ausführen.

Reply:
Schräg. Eigentlich müsste Make dann ja erkennen, dass die Dateien schon da sind, und den Download gar nicht erst starten und direkt mit den Tests weiter machen?

Du kannst die beiden Dateien auch manuell der Variable mitgeben: ein "+=" ergänzt die Liste. Und wenn Du für die beiden Dateinamen eine Variable anlegst, kannst Du die überall elegant nutzen: für die Variable C_FILES_TEST, für das Runterladen, für test_clean, ... :)

Reply:
OK, des Rätsels Lösung ist: Das Target `curl_catch` hat **gar keine** Abhängigkeiten. Das lädt einfach stumpf **immer** alles runter. 

Hier könntest Du die Variablen für die Catch-Dateien als Abhängigkeit angeben, damit das rund wird.

Also irgendwie so:

```make
CATCHFILES = $(TESTDIR)/include/catch_amalgamated.hpp  $(TESTDIR)/src/catch_amalgamated.cpp

C_FILES_TEST += $(CATCHFILES)
 
$(CATCHFILES):
	curl -o $(TESTDIR)/include/catch_amalgamated.hpp https://raw.githubusercontent.com/catchorg/Catch2/devel/extras/catch_amalgamated.hpp
	...

test: $(C_FILES_TEST)
	...

``` 

(aus der Hüfte, nicht getestet)

Reply:
@cagix stimmt, danke für den Vorschlag. Beim Testen ist mir nichts aufgefallen, hoffentlich passt es jetzt.